### ISSUE TITLE
Uploading SD file fails if original is not in root folder of 'My Content'

### ISSUE THREAD

**Issue by amarinelli** [arcpy/update-hosted-feature-service#3]
The [Upload](https://github.com/arcpy/update-hosted-feature-service/blob/e33bad62b1a21f35f2b270b75fd9fb453f5b9317/update.py#L154-L181) function produces an error and the upload fails if the original Service Definition has been moved to a folder other than the root folder in My Content of ArcGIS Online.

If the SD is moved back to the root folder and the script is re-run, it will complete successfully.

Regardless of which folder the SD is in, the item will still be found by the script and the ID printed to the screen during execution.

```python
".sd file not uploaded. Check the errors and try again."

{u'error': {u'messageCode': u'CONT_0005', u'message': u'Item does not exist in this folder.', u'code': 400, u'details': []}}
```

**Comment by amarinelli** [comment]
The reason for this error is that if an item is moved to a folder other than the root, the folder ID is required in the URL of the request.  If the item is in the root, the folder ID is removed from the URL and nothing else is required.

Example:
```
http://arcgis.com/sharing/rest/content/users/<username>/<FOLDER_ID>/items/<itemID>/update
```

**Comment by amarinelli** [comment]
Although I thought I looked for a folder property on an item, I clearlyI didn't look hard enough.  The `ownerFolder` property is `null` if the item is in the root folder or contains the folder ID if the item was moved to a folder.

Use the [Item](http://resources.arcgis.com/en/help/arcgis-rest-api/index.html#//02r30000007w000000) operation to acess this property.

My pull request #5 should likely be closed and the code reworked so that the folder ID is obtained automatically instead of the user needing to specify the containing folder (if applicable) in *settings.ini*
